static/js/subs.js

```diff
@@ -727,6 +727,48 @@ $(function () {
         update_announce_stream_state();
     });
 
+    // Search People
+    $(document).on('input', '.add-user-list-filter', function (e) {
+        var users = people.get_rest_of_realm();
+
+        var user_list = $(".add-user-list-filter");
+        if (user_list === 0) {
+            return;
+        }
+        var search_term = user_list.expectOne().val().trim();
+        var search_terms = search_term.toLowerCase().split(",");
+        search_terms = _.map(search_terms, function (s) {
+            return s.trim();
+        });
+        var filtered_users = {};
+        _.each(users, function (user) {
+            var person = people.get_by_email(user.email);
+            if (!person || !person.full_name) {
+                return false;
+            }
+            var names = person.full_name.toLowerCase().split(/\s+/);
+            names = _.map(names, function (s) {
+                return s.trim();
+            });
+            return _.any(search_terms, function (search_term) {
+                return _.any(names, function (name) {
+                    if (name.indexOf(search_term) === 0) {
```

Comment:
Nit: if you write `name.trim().indexOf(search_term.trim())` here you could get rid of two `_.map` calls above.